### ISSUE TITLE
Merging / joining meshes

### ISSUE THREAD

**Issue by 0todd0000** [mikedh/trimesh#744]
How can two overlapping meshes be merged?

The figures and scripts / code snippets below describe the problem I'm trying to solve:

- At each of three points there is a sphere of radius 0.1:
   - Point A:  (0, 0, 0)
   - Point B:  (1, 0, 0)
   - Point C:  (2, 1, 0)

I want to create a mesh that surrounds these three points, something like this:

<img width="599" alt="Screen Shot 2020-03-19 at 16 05 31" src="https://user-images.githubusercontent.com/4276626/77041425-3bd9e700-69fd-11ea-9a02-1aa6bb3a5d36.png">

- The script below uses `convex_hull` to create a mesh over all A, B and C, but not in a piecewise manner:

```python
from math import pi,ceil
import numpy as np
import trimesh


def generate_sphere_vertices(r, rad=0.1, n=36):
	theta     = np.linspace(0, pi, ceil(n/2))
	phi       = np.linspace(-pi, pi, n)
	x0,y0,z0  = r
	[t,p]     = np.meshgrid(theta, phi)
	phi,theta = t.flatten(), p.flatten()
	x         = x0 + rad * np.sin(theta) * np.cos(phi)
	y         = y0 + rad * np.sin(theta) * np.sin(phi)
	z         = z0 + rad * np.cos(theta)
	return np.vstack( [x,y,z] ).T


vertsA = generate_sphere_vertices( (0,0,0) )
vertsB = generate_sphere_vertices( (1,0,0) )
vertsC = generate_sphere_vertices( (2,1,0) )


verts   = np.vstack( [ vertsA, vertsB,vertsC ] )
pc      = trimesh.PointCloud( verts )
mesh    = pc.convex_hull
mesh.show()
```

The result looks like this:

<img width="534" alt="Screen Shot 2020-03-19 at 16 15 26" src="https://user-images.githubusercontent.com/4276626/77041353-1baa2800-69fd-11ea-88a4-9409f7e2260c.png">

To create a piecewise mesh over AB, then BC I tried to use `union`, as in the script below, but this does not work.

```python
vertsAB = np.vstack( [ vertsA , vertsB ] )
vertsBC = np.vstack( [ vertsB , vertsC ] )
pcAB    = trimesh.PointCloud( vertsAB )
pcBC    = trimesh.PointCloud( vertsBC )
meshAB  = pcAB.convex_hull
meshBC  = pcBC.convex_hull
mesh    = meshAB.union(meshBC)
mesh.show()
```

The resulting mesh looks the same as `meshAB`.

<img width="787" alt="Screen Shot 2020-03-19 at 16 22 56" src="https://user-images.githubusercontent.com/4276626/77042369-e8689880-69fe-11ea-915b-dd7879345de0.png">

Is it possible to join or merge the two meshes?

What is `union` actually doing?

**Comment by 0todd0000** [comment]
Oops!  I just noticed that this question was already answered in #245, sorry about that!

I used:

```
combined = trimesh.util.concatenate( [ meshAB, meshBC ] )
```

But I still don't understand what `union` is doing in the case above. Why doesn't `union` work?

**Comment by mikedh** [comment]
Ah, boolean operations use blender or openSCAD as a back end, and in Blender>=2.8 they substantially changed their API and it may return wrong results like this. What does `blender --version` say?

**Comment by 0todd0000** [comment]
I'm using a recent release of Blender:  version 2.82 (sub 7)
I get the same results with the openSCAD backend.

**Comment by mikedh** [comment]
Yeah, this should probably warn or fail on newer Blender versions (or we should fix it to work with the new API, PR's welcome :wink:).

SCAD for me was crashing:
```
mesh = meshAB.union(meshBC, engine='scad', debug=True)
``` 
With:
```
ERROR: CGAL error in CGALUtils::applyBinaryOperator union: CGAL ERROR: assertion violation!
Expr: itl != it->second.end()
File: /usr/include/CGAL/Nef_3/SNC_external_structure.h
Line: 1152
Current top level object is empty.
```
I'm pretty sure this is due `meshAB` not being watertight, which in turn is due to `qhull` doing something odd here.

**Comment by richardrl** [comment]
Very weird behavior with union. I get like a shadow of the objects being unioned.

![image](https://user-images.githubusercontent.com/5181990/123566778-8d3f5d00-d78e-11eb-827a-df7f21c2e47b.png)

**Comment by richardrl** [comment]
the concatenate command from above works as expected:

![image](https://user-images.githubusercontent.com/5181990/123567024-38501680-d78f-11eb-8378-b906588b9701.png)
